packages/nextjs/pages/dice.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import type { NextPage } from "next";
+import { Activities } from "~~/components/Activities";
+import { Dice } from "~~/components/Dice";
+import { MetaHeader } from "~~/components/MetaHeader";
+import { Tab } from "~~/components/Tab";
+import { useAccountBalance, useScaffoldContractRead, useScaffoldContractWrite } from "~~/hooks/scaffold-eth";
+
+const DiceGame: NextPage = () => {
+  const [currentTabIndex, setCurrentTabIndex] = useState<number>(0);
+  const rolls = [
+    { address: "0x6918da6442B0D7D4F142EB92CD6FA5288e690668", amount: 10, landedOn: "F" },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10, landedOn: "0" },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10, landedOn: "F" },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10, landedOn: "F" },
+    { address: "0xCbdAa684708c13E7b85EBe3d3410DF02F8700808", amount: 10, landedOn: "2" },
+  ];
+
+  const won = [
+    { address: "0x6918da6442B0D7D4F142EB92CD6FA5288e690668", amount: 10 },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10 },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10 },
+    { address: "0xa709FF766777C40bAC916558D9262eC4868496B4", amount: 10 },
+    { address: "0xCbdAa684708c13E7b85EBe3d3410DF02F8700808", amount: 10 },
+  ];
+
+  const { writeAsync: randomDiceRoll } = useScaffoldContractWrite({
+    contractName: "DiceGame",
+    functionName: "rollTheDice",
+  });
+
+  const onRollTypeChange = (index: number) => {
+    setCurrentTabIndex(index);
+  };
+
+  const handleRollDice = async () => {
+    if (currentTabIndex == 0) {
+      await randomDiceRoll();
+    }
+
+    /// implement the dice roll ui here
+
+    if (currentTabIndex == 1) {
+    }
+
+    /// implement the dice roll ui here
+    if (currentTabIndex == 2) {
+    }
+  };
+
+  return (
+    <>
+      <MetaHeader />
+      <div className="py-20 px-40">
+        <Tab currentIndex={currentTabIndex} onTabChange={onRollTypeChange} />
```

Comment:
I think it's not needed

Reply:
Based On the UI [Figma](https://www.figma.com/file/D0YZXzXhNtaP8j0lcq2JZR/Dice?type=design&node-id=1%3A2&mode=design&t=9nWbwZgT95vcsicq-1)